### DISCUSSION TITLE
Escaping reserved characters

### DISCUSSION THREAD

**Discussion by jobol** [mustache/spec#181] [Ideas]
Add an escaping character for protecting reserved characters. Doing thing could solve corner case issues:

- key starting with one of the characters used for section, partial, ... (`/^#<` ...)
- the question of the dot (see https://github.com/mustache/spec/pull/180)
- for possible extensions (see ex: https://gitlab.com/jobol/mustach#value-testing-equality-mustach_with_equal)

The character `\` is a good candidate.

For example, `a.b` could address the field `b` in `a` where `a\.b` address the field `a.b`.

Other example `{{\#main}}` access the field `#main`.

**Comment by agentgt** [comment]
I think this is a good idea and had similar thoughts. If we add power filters #153 I think we should consider adding some escaping. The tag names data has been kind of dangerously undefined and I think it is going to bite us in the long run. 

EDIT I will add such change IMO would have to be a 2.0 change.
